src/components/VirtualList/VirtualList.tsx

```diff
@@ -0,0 +1,71 @@
+import { useRef } from 'react';
+import { useVirtualizer } from '@tanstack/react-virtual';
+
+export interface VirtualListProps<T> {
+  data: T[];
+  renderElement: (data: T) => React.ReactNode;
+  gap?: number; // Gap in pixels between items
+  overscan?: number; // How many items to preload when scrolling
+}
+// This component is created from the example in the docs: https://tanstack.com/virtual/latest/docs/introduction
+export function VirtualList<T>({
+  data,
+  renderElement,
+  gap,
+  overscan
+}: VirtualListProps<T>) {
+  // The scrollable element for the list
+  const parentRef = useRef(null);
+
+  // The virtualizer
+  const rowVirtualizer = useVirtualizer({
+    count: data.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 35, // This is the largest estimated size tanstack seems to be able to handle
+    overscan: overscan ?? 10,
+    gap: gap ?? 5
+  });
+
+  // @TODO 2024-12-06 ylvse: style the scrollbar. Mantines scrollbar has a completely separate component for the
```

Comment:
Go through all the comments in the file and reformat them to fir within a line width of 90 characters?  Now they are all too wide